src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts

```diff
@@ -55,33 +55,42 @@ export async function fetchGenericAssayMetaByMolecularProfileIdsGroupedByGeneric
 
     const genericAssayProfilesGroupedByGenericAssayType = _.groupBy(
         genericAssayProfiles,
-        'genericAssayType'
-    );
-    const genericAssayTypes = _.keys(
-        genericAssayProfilesGroupedByGenericAssayType
+        profile => profile.genericAssayType
     );
 
-    const genericAssayMetaGroupedByGenericAssayType: {
-        [genericAssayType: string]: GenericAssayMeta[];
+    const idByGenericAssayType: {
+        [genericAssayType: string]: string[];
     } = {};
 
+    const allGenericAssayMetaMap = _.keyBy(
```

Comment:
looks to me like you can do these two "fetches" in parrelel instead of waiting for this one and then firing next.

Reply:
by next i mean the Promise.all down below.  Those aren't dependent on this one.